clang/lib/Sema/SemaDeclAttr.cpp

```diff
@@ -2958,47 +2983,171 @@ static bool checkWorkGroupSizeValues(Sema &S, Decl *D, const ParsedAttr &Attr,
   return Result;
 }
 
-// Handles reqd_work_group_size, work_group_size_hint and max_work_group_size
+template <typename WorkGroupAttrType>
+void Sema::addIntelSYCLTripleArgFunctionAttr(Decl *D,
+                                             const AttributeCommonInfo &CI,
+                                             Expr *XDim, Expr *YDim,
+                                             Expr *ZDim) {
+  assert(XDim && "Attribute must have an argument.");
+  assert(YDim && "Attribute must have an argument.");
+  assert(ZDim && "Attribute must have an argument.");
+
+  if (!XDim->isInstantiationDependent() || !YDim->isInstantiationDependent() ||
+      !ZDim->isInstantiationDependent()) {
+    Optional<llvm::APSInt> XDimVal =
+        XDim->getIntegerConstantExpr(getASTContext());
+    Optional<llvm::APSInt> YDimVal =
+        YDim->getIntegerConstantExpr(getASTContext());
+    Optional<llvm::APSInt> ZDimVal =
+        ZDim->getIntegerConstantExpr(getASTContext());
+
+    if (!XDimVal) {
+      Diag(XDim->getExprLoc(), diag::err_attribute_argument_type)
+          << CI << AANT_ArgumentIntegerConstant << XDim->getSourceRange();
+      return;
+    }
+
+    if (!YDimVal) {
+      Diag(YDim->getExprLoc(), diag::err_attribute_argument_type)
+          << CI << AANT_ArgumentIntegerConstant << YDim->getSourceRange();
+      return;
+    }
+
+    if (!ZDimVal) {
+      Diag(ZDim->getExprLoc(), diag::err_attribute_argument_type)
+          << CI << AANT_ArgumentIntegerConstant << ZDim->getSourceRange();
+      return;
+    }
+
+    int32_t XDimInt = XDimVal->getSExtValue();
```

Comment:
The old behavior was that the attribute accepted unsigned integer values, so a negative number would be treated as an unsigned value anyway. I think the new code should behave the same -- accept any integer value, and when getting the dimension values (for error checking or for codegen), get them as unsigned values.

You'd still have to check for 0 here, though.

Reply:
Thanks @AaronBallman for the feedback. 
So we will only give diagnostic for attribute value equals to 0 but not -1

Reply:
Correct -- we wouldn't diagnose negative values (because those would just be treated as positive values because the argument is unsigned), but we would still diagnose things that aren't an integer constant expression (like `1.2f`).

Reply:
Thanks for the explanation. Done.

Reply:
Done